data/filters/templates/youtube-video-duration.yaml

```diff
@@ -0,0 +1,57 @@
+  - JohnyP36
+  - Nomes77
+params:
+  - name: duration
+    description: Video duration to filter out
+    type: list
+    default: [ ]
+tags:
+  - youtube
+template: |
+  {{#each duration}}
+  {{! New style with logo, desktop }}
+  www.youtube.com##ytd-browse[page-subtype="home"] ytd-thumbnail #time-status > span:has-text(/\s({{this}})\s/):upward(ytd-rich-item-renderer)
+  www.youtube.com##ytd-browse[page-subtype="subscriptions"] ytd-thumbnail #time-status > span:has-text(/\s({{this}})\s/):upward(ytd-rich-item-renderer)
+  {{! Subscriptions in list mode, hide the whole section as subsequent videos from same channel are currently pushed in separate sections }}
+  www.youtube.com##ytd-browse[page-subtype="subscriptions"] ytd-video-renderer ytd-thumbnail #time-status > span:has-text(/\s({{this}})\s/):upward(ytd-item-section-renderer)
+  {{! Suggestion sidebar, individual shorts }}
+  www.youtube.com##ytd-watch-next-secondary-results-renderer ytd-thumbnail #time-status > span:has-text(/\s({{this}})\s/):upward(ytd-compact-video-renderer,ytd-shelf-renderer)
+  {{! Trending section }}
+  www.youtube.com##ytd-browse[page-subtype="trending"] ytd-thumbnail #time-status > span:has-text(/\s({{this}})\s/):upward(ytd-video-renderer)
+  {{! Search results }}
+  www.youtube.com##ytd-search ytd-thumbnail #time-status > span:has-text(/\s({{this}})\s/):upward(ytd-video-renderer)
+  {{/each}}
+tests:
+  - params:
+      duration: [ '0:\d\d' ]
+    output: |
+      www.youtube.com##ytd-browse[page-subtype="home"] ytd-thumbnail #time-status > span:has-text(/\s(0:\d\d)\s/):upward(ytd-rich-item-renderer)
+      www.youtube.com##ytd-browse[page-subtype="subscriptions"] ytd-thumbnail #time-status > span:has-text(/\s(0:\d\d)\s/):upward(ytd-rich-item-renderer)
+      www.youtube.com##ytd-browse[page-subtype="subscriptions"] ytd-video-renderer ytd-thumbnail #time-status > span:has-text(/\s(0:\d\d)\s/):upward(ytd-item-section-renderer)
+      www.youtube.com##ytd-watch-next-secondary-results-renderer ytd-thumbnail #time-status > span:has-text(/\s(0:\d\d)\s/):upward(ytd-compact-video-renderer,ytd-shelf-renderer)
+      www.youtube.com##ytd-browse[page-subtype="trending"] ytd-thumbnail #time-status > span:has-text(/\s(0:\d\d)\s/):upward(ytd-video-renderer)
+      www.youtube.com##ytd-search ytd-thumbnail #time-status > span:has-text(/\s(0:\d\d)\s/):upward(ytd-video-renderer)
+---
+
+With this filter format it is possible to filter out videos with any duration. This filter hides most
+occurrences of this format:
+
+- videos in the subscription and search pages
+- dedicated shelves in the homepage and channel list
+
+The input should be a regexp. In regexp '\d' is for every possible digit. Multiple options should be divided be a '|', some examples:
+
+- less than a minute: 0:\d\d
+- between one and two minutes: 1:\d\d
+- one minute or more: 1:\d\d|\d:\d\d|\d\d:\d\d|\d\d:\d\d:\d\d
+- less than 10 minutes: \d:\d\d
+- 10 minutes or more: 1\d:\d\d|\d:\d\d:\d\d|\d\d:\d\d:\d\d
```

Comment:
What if people want everything between 10 and 25 minutes: `1\d:\d\d|20:\d\d|21:\d\d|22:\d\d|23:\d\d|24:\d\d` would be long. Is there a way to make that shorter?

Reply:
Yes: `1\d:\d\d|(20|21|22|23|24):\d\d`

Reply:
or shorter: `1\d:\d\d|2[0-5]:\d\d`

Reply:
`1\d:\d\d` will block 10:00-19:59; `\d:\d\d:\d\d` will block 1:00:00-9:59:59; `\d\d:\d\d:\d\d` will block 10:00:00-99:59:59.
`\d` means 0-9.

Reply:
Thus, not between 20:00-59:59 ???

Reply:
Yes, but see https://github.com/letsblockit/letsblockit/pull/622/commits/777e36d266310ec3cd0dafb35563c69c009bbeac

Maybe we should add a link to a website with a good explenation to all regexp syntax rules?

Reply:
> Maybe we should add a link to a website with a good explenation to all regexp syntax rules?

Probably.

But check line 50
https://github.com/letsblockit/letsblockit/blob/de93b3eca28d4452342b4f3e703c5433870cad72/data/filters/templates/youtube-video-duration.yaml#L50

After the first `\d` is an ) which doesn't have a (. Should it be removed?

Reply:
Yes, done: https://github.com/letsblockit/letsblockit/pull/622/commits/8a7edb6b892643844e403f79e399d221a3326c40

Reply:
Completely fixed in https://github.com/letsblockit/letsblockit/pull/622/commits/87716edc05499e8df7cf677fe6b52f4dbaf2b767